src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -299,7 +299,11 @@ def resolve(self, org_id: int, string: str) -> Optional[int]:
 
         metrics.incr(_INDEXER_CACHE_METRIC, tags={"cache_hit": "false", "caller": "resolve"})
         try:
-            id: int = StringIndexerTable.objects.get(organization_id=org_id, string=string).id
+            id: int = (
+                StringIndexerTable.objects.using_replica()
+                .get(organization_id=org_id, string=string)
+                .id
+            )
```

Comment:
Nit: You only want one value here, so it could make sense to use `.filter(id).values_list('id', flat=True)[0]` instead and catch the `IndexError`.